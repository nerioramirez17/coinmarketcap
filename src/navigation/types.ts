import { RouteProp } from '@react-navigation/native';
import { Cryptocurrency } from '../api/types';
import { StackNavigationProp } from '@react-navigation/stack';


export type RootStackParams = {
  Login: undefined;
  Home: undefined;
  Detail: { item: Cryptocurrency };
};

type DetailScreenRouteProp = RouteProp<RootStackParams, 'Detail'>;
export type LoginScreenNavigationProp = StackNavigationProp<RootStackParams, 'Login'>;
export type HomeScreenNavigationProp = StackNavigationProp<RootStackParams, 'Home'>;
export type DetailProps = {
  route: DetailScreenRouteProp;
};

