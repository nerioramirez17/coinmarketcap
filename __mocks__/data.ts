import { Cryptocurrency } from '../src/api/types';
import { User } from '@react-native-google-signin/google-signin';

const currenciesData: Cryptocurrency[] = [
  {
    id: 1,
    name: 'Bitcoin',
    symbol: 'BTC',
    num_market_pairs: 10943,
    date_added: '2010-07-13T00:00:00.000Z',
    max_supply: 21000000,
    circulating_supply: 19664912,
    total_supply: 19664912,
    infinite_supply: false,
    last_updated: '2024-03-26T16:46:00.000Z',
    quote: {
      USD: {
        price: 69599.71303641264,
        volume_24h: 40664759292.57396,
        percent_change_1h: -1.03136701,
        percent_change_24h: -0.73631703,
        market_cap: 1368672232086.3074,
        market_cap_dominance: 52.24,
        fully_diluted_market_cap: 1461593973764.67,
        last_updated: '2024-03-26T16:46:00.000Z',
      },
    },
  },
  {
    id: 1027,
    name: 'Ethereum',
    symbol: 'ETH',
    num_market_pairs: 8652,
    date_added: '2015-08-07T00:00:00.000Z',
    max_supply: null,
    circulating_supply: 120074016.27550143,
    total_supply: 120074016.27550143,
    infinite_supply: true,
    last_updated: '2024-03-26T16:45:00.000Z',
    quote: {
      USD: {
        price: 3549.137750052892,
        volume_24h: 20406590605.586422,
        percent_change_1h: -1.23818139,
        percent_change_24h: -1.37018086,
        market_cap: 426159223963.8475,
        market_cap_dominance: 16.2231,
        fully_diluted_market_cap: 426159223963.85,
        last_updated: '2024-03-26T16:45:00.000Z',
      },
    },
  },
];

const userData: User = {
  idToken: '13456-ID',
  serverAuthCode: '1234',
  user: {
    email: 'nerio@challenge.com',
    familyName: 'Testing',
    givenName: 'Challenge',
    id: '1',
    name: 'test',
    photo: 'https://www.test.com/photo.png',
  },
};


export { currenciesData, userData };
