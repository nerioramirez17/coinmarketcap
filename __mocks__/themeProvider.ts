
const mockTheme = {
  background: {
    secondary: '#FFF', 
  },
  text: {
    primary: '#FFF'
  }
  
};

export default mockTheme;
