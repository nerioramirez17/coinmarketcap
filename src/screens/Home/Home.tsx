import React, { FC, useMemo, useState } from 'react';
import { GoogleSignin } from '@react-native-google-signin/google-signin';
import { FlatList, Pressable, RefreshControl } from 'react-native';
import {  
  CardContainer,
  Container,
  Icon,
  IconContainer,
  Layout,
  NumbersContainer,
  PercentText,
  PressableIconLike,
  TextContainer,
  TextFav,
  TitleName,
  TitlePrice,
  TitleSymbol,
  Toggle,
  ToggleContainer,
  ButtonContainer,
  ButtonText,
  ToggleButtonContainer,
} from './styles';
import { useQuery } from 'react-query';
import { useApi } from '../../api/';
import { useFavorites } from '../../hooks/useFavorites';
import { useUser } from '../../hooks/useUser';
import { Cryptocurrency } from '../../api/types';
import { HomeScreenNavigationProp } from '../../navigation/types';
import LikeIcon from '../../../assets/Icons/like.svg';
import UnLikeIcon from '../../../assets/Icons/unlike.svg';
import { Search } from '../../components/search/Search';
import { theme  } from '../../theme/';

const URI_ICON = 'https://s2.coinmarketcap.com/static/img/coins/64x64/';

type ItemProps =  {
  item: Cryptocurrency;
};

type RenderItemProps = {
  item : Cryptocurrency
};

const Home = ({ navigation }: { navigation: HomeScreenNavigationProp }) => {
  const [filterCriteria, setFilterCriteria] = useState<string>('');
  const [seeFavorites, setSeeFavotes] = useState(false);
  const { addFavorite, isFavorite, removeFavorite, favorites } = useFavorites();
  const { getCurrencies } = useApi();
  const { clearUser } = useUser();
  const { data, refetch, isFetching } = useQuery('currencies', getCurrencies, {
    refetchInterval: 30000,
  });


  const matchesFilterCriteria = (item: any, criteria: string) => {
    const lowerCaseCriteria = criteria.toLowerCase();
    return (
      item.name.toLowerCase().includes(lowerCaseCriteria) ||
      item.symbol.toLowerCase().includes(lowerCaseCriteria)
    );
  };

  const filteredData = useMemo(
    () =>
      data?.filter(
        item =>
          matchesFilterCriteria(item, filterCriteria),
      ),
    [data, filterCriteria],
  );

  const matchesFavorites = filteredData?.filter(item => favorites.includes(item.symbol));
  
  const formatPrice = (price: number) => {
    return price?.toFixed(2);
  };

  const onRefresh = () => {
    refetch();
  };

  const renderItem = ({ item }: RenderItemProps) => (
    <Item
      key={item.id}
      item={item}
    />
  );

  const signOut = async () => {
    try {
      await GoogleSignin.revokeAccess();
      await GoogleSignin.signOut();
      clearUser();
      navigation.navigate('Login');
    } catch (error) {
      console.error(error);
    }
  };

  const Item: FC<ItemProps> = ({
   item
  }) => {    
    const isLiked = useMemo(() => isFavorite(item.symbol), [isFavorite, item.symbol]);    
    const addFavorites = (() => {
      isLiked ? removeFavorite(item.symbol) : addFavorite(item.symbol);
    })
    
    return (
      <Pressable onPress={() => { navigation.navigate('Detail', { item })}}>
      <CardContainer>
        <IconContainer>
          <Icon source={{ uri: `${URI_ICON}${item.id}.png` }} />
          <TextContainer>
            <TitleName>{item.name}</TitleName>
            <TitleSymbol>{item.symbol}</TitleSymbol>
          </TextContainer>
        </IconContainer>
       <NumbersContainer>
           <TitlePrice>{formatPrice(item.quote?.USD.price)}</TitlePrice>
          <PercentText negative={item.quote?.USD.percent_change_24h < 0}>
            {item.quote?.USD.percent_change_24h}%
          </PercentText>  
       </NumbersContainer>
       <PressableIconLike onPress={addFavorites}>
          {
            isLiked ? <LikeIcon width={12} height={12}/> : <UnLikeIcon width={16} height={16}/>
          }
        </PressableIconLike> 
      </CardContainer>
      </Pressable>

    );
  };

  return (
    <Layout>
      <Container>
        <Search setFilterCriteria={setFilterCriteria}/>
        <ToggleButtonContainer>
          <ToggleContainer>
            <Toggle
              trackColor={{ false: theme.button.tertiary, true: theme.button.primary }}
              ios_backgroundColor={theme.background.inversed}
              onValueChange={() =>  setSeeFavotes(previousState => !previousState)}
              value={seeFavorites}
              />
            <TextFav>Favoritos</TextFav>
          </ToggleContainer>
        <ButtonContainer onPress={signOut}>
          <ButtonText>Cerrar Sesion</ButtonText>
        </ButtonContainer>
        </ToggleButtonContainer>
          <FlatList
            data={seeFavorites ? matchesFavorites : filteredData}
            renderItem={renderItem}
            contentContainerStyle={{ paddingHorizontal: 16 }}
            showsVerticalScrollIndicator={false}
            refreshControl={
              <RefreshControl refreshing={isFetching} onRefresh={onRefresh} />
            }
          />
      </Container>
    </Layout>
  );
};


export { Home };