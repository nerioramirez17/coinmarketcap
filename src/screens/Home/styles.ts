import { Image, Pressable, Switch, Text, TouchableOpacity, View } from 'react-native';
import styled from 'styled-components/native';

export const Layout = styled(View)`
  background-color: ${({ theme }) => theme.background.secondary};
  flex: 1;
`;

export const CardContainer = styled(View)`
  background-color: ${({ theme }) => theme.background.primary};
  border-radius: 8px;
  padding: 12px;
  margin: 6px;
  flex-direction: row;
  justify-content: space-between;
`;

export const IconContainer = styled(View)`
  flex-direction: row;
  margin-bottom: 14px;
`;

export const TextContainer = styled(View)`
  margin-left: 12px;
`;

export const NumbersContainer = styled(View)`
  margin-left: 12px;
  margin-right: 22px;
`;

export const Icon = styled(Image)`
  width: 40px;
  height: 40px;
  border-radius: 20px;
`;

export const IconLike = styled(Image)`
  width: 16px;
  height: 16px;
`;

export const TitleName = styled(Text)`
  font-size: 16px;
  color: ${({ theme }) => theme.text.primary};
  font-weight: bold;
`;

export const TitleSymbol = styled(Text)`
  font-size: 14px;
  color: ${({ theme }) => theme.text.tertiary};
`;

export const TitlePrice = styled(Text)`
  font-size: 14px;
  text-align: right;
  color: ${({ theme }) => theme.text.tertiary};
`;

export const PercentText = styled(Text)<{ negative: boolean }>`
  font-size: 12px;
  text-align: right;
  color: ${({ negative, theme }) =>
    negative ? theme.text.negative : theme.text.positive};
  font-weight: 400;
`;

export const PressableIconLike = styled(Pressable)`
  position: absolute;
  right: 8px;
  top: 6px;
`;

export const ToggleContainer = styled(View)`
  margin-left: 22px;
  margin-bottom: 16px;
  margin-top: 16px;
`;

export const ToggleButtonContainer = styled(View)`
  flex-direction: row;
  justify-content: space-between;
`;

export const Toggle = styled(Switch)`
  margin-bottom: 8px;
`;

export const ButtonContainer = styled(TouchableOpacity)`
  margin: 30px;
  justify-content: center;
  align-items: center;
  border-radius: 18px;
  width: 160px;
  background-color: ${({ theme }) => theme.background.primary};
`;

export const ButtonText = styled(Text)`
  font-size: 12px;
  color: ${({ theme }) => theme.text.primary};
`;

export const TextFav = styled(Text)`
  margin-bottom: 8px;
  font-weight: 600;
  color: ${({ theme }) => theme.text.primary};
`;


export const Container = styled(View)`
  flex: 1;
  margin-top: 50px
`;




