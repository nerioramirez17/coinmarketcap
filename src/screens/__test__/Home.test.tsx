import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { Home } from '../Home/Home';
import { ThemeProvider } from 'styled-components/native';
import mockTheme from '../../../__mocks__/themeProvider';
import { currenciesData, userData } from '../../../__mocks__/data';
import { mockNavigation } from '../../../__mocks__/navigation';


const mockData = { data: currenciesData };
const mockUserData = { data: userData };
jest.mock('react-query', () => ({
  ...jest.requireActual('react-query'),
  useQuery: () => mockData,
}));

jest.mock('../../hooks/useUser', () => ({
  useUser: jest.fn(() => ({
    clearUser: jest.fn(),
    user: mockUserData,
  })),
}));

jest.mock('../../hooks/useFavorites', () => ({
  useFavorites: () => ({
    addFavorite: jest.fn(),
    isFavorite: jest.fn(),
    removeFavorite: jest.fn(),
    favorites: [],
  }),
}));

const renderHomeComponent = () => {
  return render(
      <ThemeProvider theme={mockTheme}>
        <Home navigation={mockNavigation} />
      </ThemeProvider>
  );
};

describe('Home Component', () => {
  it('should render a snapshot with data', () => {
    const { toJSON } = renderHomeComponent();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should navigate when Item press', () => {
    const { getByText } = renderHomeComponent();
    fireEvent.press(getByText('Bitcoin'));
    expect(mockNavigation.navigate).toHaveBeenCalledWith('Detail', {
      item: currenciesData[0],
    });
  });
});
