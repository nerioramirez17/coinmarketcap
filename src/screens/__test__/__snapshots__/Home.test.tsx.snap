// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render a snapshot with data 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFF",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginTop": 50,
      }
    }
  >
    <View
      style={
        {
          "borderBottomLeftRadius": 28,
          "borderBottomRightRadius": 28,
          "borderColor": "#FFF",
          "borderTopLeftRadius": 28,
          "borderTopRightRadius": 28,
          "borderWidth": 1,
          "marginBottom": 16,
          "marginLeft": 16,
          "marginRight": 16,
          "marginTop": 16,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <SvgMock
          height={18}
          style={
            {
              "marginRight": 8,
            }
          }
          width={18}
        />
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          autoFocus={true}
          onChangeText={[Function]}
          placeholder="Buscar Criptomonedas"
          placeholderTextColor="#6C757D"
          style={
            {
              "color": "#FFF",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            }
          }
          value=""
        />
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          {
            "marginBottom": 16,
            "marginLeft": 22,
            "marginTop": 16,
          }
        }
      >
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="#2FA3FF"
          style={
            [
              {
                "height": 31,
                "width": 51,
              },
              [
                {
                  "marginBottom": 8,
                },
                {
                  "backgroundColor": "#D3D3D3",
                  "borderRadius": 16,
                },
              ],
            ]
          }
          tintColor="#767577"
          value={false}
        />
        <Text
          style={
            {
              "color": "#FFF",
              "fontWeight": "600",
              "marginBottom": 8,
            }
          }
        >
          Favoritos
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "",
            "borderBottomLeftRadius": 18,
            "borderBottomRightRadius": 18,
            "borderTopLeftRadius": 18,
            "borderTopRightRadius": 18,
            "justifyContent": "center",
            "marginBottom": 30,
            "marginLeft": 30,
            "marginRight": 30,
            "marginTop": 30,
            "opacity": 1,
            "width": 160,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          Cerrar Sesion
        </Text>
      </View>
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "paddingHorizontal": 16,
        }
      }
      data={
        [
          {
            "circulating_supply": 19664912,
            "date_added": "2010-07-13T00:00:00.000Z",
            "id": 1,
            "infinite_supply": false,
            "last_updated": "2024-03-26T16:46:00.000Z",
            "max_supply": 21000000,
            "name": "Bitcoin",
            "num_market_pairs": 10943,
            "quote": {
              "USD": {
                "fully_diluted_market_cap": 1461593973764.67,
                "last_updated": "2024-03-26T16:46:00.000Z",
                "market_cap": 1368672232086.3074,
                "market_cap_dominance": 52.24,
                "percent_change_1h": -1.03136701,
                "percent_change_24h": -0.73631703,
                "price": 69599.71303641264,
                "volume_24h": 40664759292.57396,
              },
            },
            "symbol": "BTC",
            "total_supply": 19664912,
          },
          {
            "circulating_supply": 120074016.27550143,
            "date_added": "2015-08-07T00:00:00.000Z",
            "id": 1027,
            "infinite_supply": true,
            "last_updated": "2024-03-26T16:45:00.000Z",
            "max_supply": null,
            "name": "Ethereum",
            "num_market_pairs": 8652,
            "quote": {
              "USD": {
                "fully_diluted_market_cap": 426159223963.85,
                "last_updated": "2024-03-26T16:45:00.000Z",
                "market_cap": 426159223963.8475,
                "market_cap_dominance": 16.2231,
                "percent_change_1h": -1.23818139,
                "percent_change_24h": -1.37018086,
                "price": 3549.137750052892,
                "volume_24h": 20406590605.586422,
              },
            },
            "symbol": "ETH",
            "total_supply": 120074016.27550143,
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
        />
      }
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <RCTRefreshControl />
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                {
                  "backgroundColor": "",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 6,
                  "marginLeft": 6,
                  "marginRight": 6,
                  "marginTop": 6,
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                    "marginBottom": 14,
                  }
                }
              >
                <Image
                  source={
                    {
                      "uri": "https://s2.coinmarketcap.com/static/img/coins/64x64/1.png",
                    }
                  }
                  style={
                    {
                      "borderBottomLeftRadius": 20,
                      "borderBottomRightRadius": 20,
                      "borderTopLeftRadius": 20,
                      "borderTopRightRadius": 20,
                      "height": 40,
                      "width": 40,
                    }
                  }
                />
                <View
                  style={
                    {
                      "marginLeft": 12,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#FFF",
                        "fontSize": 16,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Bitcoin
                  </Text>
                  <Text
                    style={
                      {
                        "color": "",
                        "fontSize": 14,
                      }
                    }
                  >
                    BTC
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "marginLeft": 12,
                    "marginRight": 22,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "",
                      "fontSize": 14,
                      "textAlign": "right",
                    }
                  }
                >
                  69599.71
                </Text>
                <Text
                  negative={true}
                  style={
                    {
                      "color": "",
                      "fontSize": 12,
                      "fontWeight": "400",
                      "textAlign": "right",
                    }
                  }
                >
                  -0.73631703
                  %
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "position": "absolute",
                    "right": 8,
                    "top": 6,
                  }
                }
              >
                <SvgMock
                  height={16}
                  width={16}
                />
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                {
                  "backgroundColor": "",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 6,
                  "marginLeft": 6,
                  "marginRight": 6,
                  "marginTop": 6,
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                    "marginBottom": 14,
                  }
                }
              >
                <Image
                  source={
                    {
                      "uri": "https://s2.coinmarketcap.com/static/img/coins/64x64/1027.png",
                    }
                  }
                  style={
                    {
                      "borderBottomLeftRadius": 20,
                      "borderBottomRightRadius": 20,
                      "borderTopLeftRadius": 20,
                      "borderTopRightRadius": 20,
                      "height": 40,
                      "width": 40,
                    }
                  }
                />
                <View
                  style={
                    {
                      "marginLeft": 12,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#FFF",
                        "fontSize": 16,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Ethereum
                  </Text>
                  <Text
                    style={
                      {
                        "color": "",
                        "fontSize": 14,
                      }
                    }
                  >
                    ETH
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "marginLeft": 12,
                    "marginRight": 22,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "",
                      "fontSize": 14,
                      "textAlign": "right",
                    }
                  }
                >
                  3549.14
                </Text>
                <Text
                  negative={true}
                  style={
                    {
                      "color": "",
                      "fontSize": 12,
                      "fontWeight": "400",
                      "textAlign": "right",
                    }
                  }
                >
                  -1.37018086
                  %
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "position": "absolute",
                    "right": 8,
                    "top": 6,
                  }
                }
              >
                <SvgMock
                  height={16}
                  width={16}
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
